plugins/lookup/bitwarden.py

```diff
@@ -161,37 +250,83 @@ def _get_matches(self, search_value, search_field, collection_id):
+
         matches = self._get_matches(search_value, search_field, collection_id)
+        if not matches and edit_record:
+            return [self._create_record(search_value, field, collection_id, generate, set_)]
+
         if not field:
             return matches
         field_matches = []
         for match in matches:
-            # if there are no custom fields, then `match` has no key 'fields'
-            if 'fields' in match:
-                custom_field_found = False
-                for custom_field in match['fields']:
-                    if field == custom_field['name']:
-                        field_matches.append(custom_field['value'])
-                        custom_field_found = True
-                        break
-                if custom_field_found:
+            display.vvvv("Looking in match %s for field %s..." % (match.get('name'), field))
+            if edit_record:
+                field_matches.append(self._check_and_edit_match(field, generate, match, set_))
+            else:
+                # if there are no custom fields, then `match` has no key 'fields'
+                if 'fields' in match:
+                    custom_field_found = False
+                    for custom_field in match['fields']:
+                        if field == custom_field['name']:
+                            field_matches.append(custom_field['value'])
+                            custom_field_found = True
+                            break
+                    if custom_field_found:
+                        continue
+                if 'login' in match and field in match['login']:
+                    field_matches.append(match['login'][field])
+                    continue
+                if field in match:
+                    field_matches.append(match[field])
                     continue
-            if 'login' in match and field in match['login']:
-                field_matches.append(match['login'][field])
-                continue
-            if field in match:
-                field_matches.append(match[field])
-                continue
         if matches and not field_matches:
             raise AnsibleError("field {field} does not exist in {search_value}".format(field=field, search_value=search_value))
         return field_matches
 
+    def _check_and_edit_match(self, field, generate, match, set_):
+        if field in ['username', 'password', 'uris']:
+            if 'login' not in match:
+                match['login'] = {}
+            if field == 'uris':
+                for uri in match['login'].get('uris', []):
+                    if uri.get('uri') == set_:
+                        return set_
+                value = self._generate_password(generate) if set_ is None else set_
+                match['login']['uris'] = match['login'].get('uris', []) + [{'uri': value}]
+                self._edit_record(match['id'], match)
+                display.v("  added uri=%s ..." % value)
```

Comment:
Logging a secret value is something that should never be done. You can either do
```suggestion
                display.v("  added uri ...")
```
or
```suggestion
```

Reply:
`set` should not be logged, as it is a secret.
`generate` is autogenerated, so we don't care about the value and it also does not need to be logged.

-> removed logging of any value